src/tests/ftest/daos_test/daos_core_test.py

```diff
@@ -23,57 +23,24 @@
 """
 
 import os
-import sys
-import json
 import subprocess
-
-from avocado       import Test
 from avocado.utils import process
+from apricot import TestWithServers
 
-sys.path.append('./util')
-sys.path.append('../util')
-sys.path.append('../../../utils/py')
-sys.path.append('./../../utils/py')
-# pylint: disable=wrong-import-position,import-error
-import AgentUtils
-import server_utils
-import write_host_file
-# pylint: enable=wrong-import-position,import-error
-
-class DaosCoreTest(Test):
+class DaosCoreTest(TestWithServers):
     """
     Runs the daos_test subtests with multiple servers.
-
+    :avocado: recursive
     """
-    # pylint: disable=too-many-instance-attributes
     subtest_name = None
 
     def setUp(self):
-        self.agent_sessions = None
-        self.hostlist = None
+        super(DaosCoreTest, self).setUp()
         self.subtest_name = self.params.get("test_name",
                                             '/run/daos_tests/Tests/*')
 
-        # get paths from the build_vars generated by build
-        with open('../../../.build_vars.json') as build_vars:
-            build_paths = json.load(build_vars)
-        self.basepath = os.path.normpath(build_paths['PREFIX'] + "/../")
-
-        self.server_group = self.params.get("server_group", '/server/',
-                                            'daos_server')
-        self.daos_test = self.basepath + '/install/bin/daos_test'
```

Comment:
aren't some of these unique to this test?

Reply:
Not entirely sure what you mean, but I don't think so.  This test does run and pass with this patch.